dask/_task_spec.py

```diff
@@ -740,6 +754,15 @@ def __call__(self, values=()):
             return self.func(*new_argspec, **kwargs)
         return self.func(*new_argspec)
 
+    def __setstate__(self, state):
+        slots = self.__class__.get_all_slots()
+        for sl, val in zip(slots, state):
+            setattr(self, sl, val)
+
+    def __getstate__(self):
+        slots = self.__class__.get_all_slots()
+        return tuple(getattr(self, sl) for sl in slots)
```

Comment:
This is an implementation that should cover all subclasses that do not require any custom logic upon deserialization and are implementing the slots protocol properly, i.e. subclasses typically don't have to implement anything else